spec/requests/clean_paths_spec.rb

```diff
@@ -12,4 +12,12 @@
 
     expect(response.header['Location']).to eq('/about/faq?test=true')
   end
+
+  it 'do not strip .xml extension for sitemap.xml' do
+    get 'http://example.com/sitemap.xml'
+
+    expect(response.status).to eq(200)
+    expect(response.header['Location']).to eq(nil)
+    expect(response.body).to eq('<sitemap>test</sitemap>')
+  end
```

Comment:
Maybe instead, we need to run the rake tasks that you mention in the PR description.

Reply:
This is pretty clean and it make me realize that the `clean_paths` middleware does not remove the extension since the sitemap is compressed and the extension is `.gz` and not `.xml`.